### ISSUE TITLE
SubX CFSv2 data access

### ISSUE THREAD

**Issue by btickell** [microsoft/subseasonal_data#2]
I am trying to access the SubX CFSv2 temperature forecasts used in this project in order to do some slightly different preprocessing and am running into an error with the netCDF4 back-end of xarray. 

```
import xarray as xa
x = xa.open_dataset(
  "https://iridl.ldeo.columbia.edu/SOURCES/.Models/.SubX/.NCEP/.CFSv2/.forecast/.tas/dods",
    decode_times=False,
)
x['tas'][0, :, :, :, :].data 
```

causes the following error

```
File src/netCDF4/_netCDF4.pyx:4406, in netCDF4._netCDF4.Variable.__getitem__()

File src/netCDF4/_netCDF4.pyx:5348, in netCDF4._netCDF4.Variable._get()

IndexError: index exceeds dimension bounds
```
This also occurs when simply trying to access the data as `x['tas'].data`

Is this the way you access the data source for this project? If not would you mind detailing how you actually download the source files?

Thanks!

**Comment by lmackey** [comment]
Dear Blake,

Our download process is a bit different.  We download one day's worth of
data each day by

   1. Constructing the appropriate URL for a day's worth of data using the
   following template
      - daily_url_template = "
      http://iridl.ldeo.columbia.edu/SOURCES/.Models/.SubX/.{url_suffix}/.{model_name}/.{outlook}/.{variable}/S/(0000%20{day}%20{month}%20{year})(1800%20{day}%20{month}%20{year})RANGE/X/({})({})RANGEEDGES/Y/({})({})RANGEEDGES/[M]average/data.nc".format(*long_range,
      *lat_range)
   2. Using the requests package to fetch the data from the URL
      - import requests
      - r = requests.get(url, timeout=timeout)
   3. Writing the data to file
      - with open(file_name, 'wb') as f:
         - f.write(r.content)

In case we have already downloaded relevant data, what sorts of
preprocessing changes are you hoping to make?

**Comment by btickell** [comment]
Thank you, this is actually the approach I more or less ended up with. I was seeking an understanding of the processing pipeline going from the SubX source to the dataframe your project provides so this was very helpful. 

We were mostly looking for daily, non-averaged leads but were unable to access the data sources through the official scripts provided by the SubX project.

**Comment by btickell** [comment]
I am interested in one fairly small detail. The leads available from this source go to 43.5 days from the issuance date. You specify that you average the 14 leads starting at l for each lead. This seems to limit the number of leads in this method to 29.5.

This checks out with the dataframe made available. I am wondering how you take issuance date d, l and date d-1, l+1 for the 29.5 column. Do you take 13 days average from the 30.5 lead and use that?

**Comment by lmackey** [comment]
Dear Blake,

The initial step of averaging over 14 leads limits the final lead to 30.5,
as the outputted 30.5 value becomes the average over the 14 leads 30.5,
..., 43.5.  Then, for each issuance date t, lead l, and grid point, we
replace the forecast with the average of the (issuance date t, lead l) and
(issuance date t-1, lead l+1) forecasts.  For lead 29.5, this amounts to
taking the average of (issuance date t, lead 29.5) and (issuance date t-1,
lead 30.5).  Please let us know if you have any additional questions!
